pyanaconda/localization.py

```diff
@@ -119,47 +119,73 @@ def find_best_locale_match(locale, langcodes):
     :return: the best matching langcode from the list of None if none matches
     :rtype: str or None
     """
-    def score_value_pair(locale_value, langcode_value, weight):
-        if locale_value and langcode_value:
-            if locale_value == langcode_value:
-                # match
-                return weight
-            else:
-                # not match
-                return -weight
-        elif langcode_value and not locale_value:
-            # langcode has something the locale doesn't have
-            return -weight
-        return 0
-
+    # Parse the locale.
     if not is_valid_langcode(locale):
         return None
 
-    scores = []
+    locale_parsed = langtable.parse_locale(locale)
+
+    # Get a score for each langcode.
+    scores = {}
 
-    # get score for each langcode
     for langcode in langcodes:
         if not is_valid_langcode(langcode):
-            scores.append((langcode, 0))
+            continue
+
+        langcode_parsed = langtable.parse_locale(langcode)
+
+        # Don't match a non-POSIX locale with a POSIX langcode.
+        if langcode_parsed.variant == "POSIX" and locale_parsed.variant != "POSIX":
+            continue
+
+        score = _evaluate_locales(locale_parsed, langcode_parsed)
+
+        # Matches matching only script or encoding or both are not useful.
+        # The score of 100 requires at least territory to have matched.
+        if score <= 100:
+            continue
+
+        scores[langcode] = score
```

Comment:
I just wonder if there couldn't be duplicate langcode. In that case the original solution would keep them separate. On the other hand even if they would they should have the same score so it shouldn't matter?

Reply:
Duplicate langcodes will always have the same score.